### ISSUE TITLE
Opaque identifier in injectedWeb3

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/extension#1174]
Copied from elsewhere: (which is a summary of an offline report)

The `window.injectedWeb3` object does "leak" information about what is available to the specific user. When everybody plays nice, not an issue, however it does show all sites what is there without the user even having taken action to enable/disable. So this needs to be more opaque until `enable` (or something like enable) indicates stuff is approved by the user and the underlying info is returned.

- [ ] don't leak name/version info before enabled
- [ ] opaque identifiers on a per-load basis
- [ ] backwards compatible for older extensions (critical on dapp layer)

This most-probably means a new `connect(<name>)` interface that return the name/version with the older `enable(<name>)` available for non-current versions.

TL;DR Needs to be fleshed out a bit...
